docs/source/user_guide/pipelines.md

```diff
@@ -66,167 +66,32 @@ needed for the pipeline are present.
-					"port_id_ref": "outPort"
-				}, {
-					"node_id_ref": "f6584209-6f22-434f-9820-41327b6c749d",
-					"port_id_ref": "outPort"
-				}]
-			}],
-			"outputs": [{
-				"id": "outPort",
-				"app_data": {
-					"ui_data": {
-						"label": ""
-					}
-				}
-			}]
-		}],
-		"app_data": {
-			"ui_data": {
-				"comments": []
-			},
-			"title": "pipeline-title",
-			"runtime": "kfp",
-			"runtime-config": "kfp-yukked1"
-		},
-		"runtime_ref": ""
-	}],
-	"schemas": []
-}
-```
-
-## Pipeline processor design
-
-Elyra implements an extensible **pipeline processor engine**, which enables the addition of new processors utilizing
-a service discovery mechanism.
-
-![Notebook Pipeline Editor](../images/pipeline-class-hierarchy.png)
-
-
-### Registering new pipeline processor
-
-New processors that properly implement the **Pipeline Processor** service definition can be discovered
-by the runtime using **entry_points** defined in the **setup.py** file (either from Elyra or in the package
-where the processor is located):
-
-```python
-    entry_points={
-        'elyra.pipeline.processors': [
-            'kfp = elyra.pipeline.processor_kfp:KfpPipelineProcessor'
-        ]
-    },
-```
+### Distributing Your Pipeline
+Oftentimes you'll want to share or distribute your pipeline (including its notebooks and their dependencies) with colleagues.  This section covers some of the best practices for accomplishing that, but first, it's good to understand the relationships between components of a pipeline.
+
+#### Pipeline Component Relationships
+The primary component of a pipeline is the pipeline file itself.  This JSON file (with a `.pipeline` extension) contains all relationships of the pipeline.  The notebook execution nodes each specify a notebook file (a JSON file with a `.ipynb` extension) who's path is currently **relative to the working directory of the hosting notebook server**.  Each dependency of a given node is relative to the notebook location itself - **not** the server's working directory.  When a pipeline is submitted for processing or export, the pipeline file itself is not sent to the server, only a portion of its contents are sent.
+
+#### Distributing Pipelines - Best Practice
+Prior to distributing your pipeline - which includes preserving the component relationships - it is best to commit these files (and directories) to a github repository.  An alternative approach would be to archive the files using `tar` or `zip`, but, again preserving the component relationships.
+
+When deploying a shared or distributed pipeline repository or archive, it is **very important** that the pipeline notebooks be extracted into the **same directory hierarchy relative to the hosting notebook server's working directory**.
+
+
+For example, if you archived your pipeline into a github repository named `my_analytics_pipeline` from the notebook server's working directory, then consumers of this archive would need to do the following:
+
+1. Clone your repository
```

Comment:
I believe there are two options/best practices here: 
1 - The one you describe, which requires one single project per workspace
2 - publishing your project inside a folder, probably the same name as the repo, which should allow us to have multiple github projects in one workspace

Thoughts?  Should we also mention the number 2 ?

Reply:
Yes - I agree.  It's all a matter of capturing the project relative to the notebook directory.  Thanks for the suggestion.